app/javascript/components/map/components/basemaps/basemaps-component.jsx

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Dropdown from 'components/ui/dropdown';
+import cx from 'classnames';
+import Icon from 'components/ui/icon';
+import closeIcon from 'assets/icons/close.svg';
+import infoIcon from 'assets/icons/info.svg';
+
+import basemaps, { labels } from './basemaps-schema';
+import './styles.scss';
+
+class Basemaps extends React.PureComponent {
+  static propTypes = {
+    onClose: PropTypes.func.isRequired,
+    boundaries: PropTypes.array.isRequired,
+    selectLabels: PropTypes.func.isRequired,
+    selectBasemap: PropTypes.func.isRequired,
+    activeLabels: PropTypes.object.isRequired,
+    activeBasemap: PropTypes.object.isRequired,
+    selectBoundaries: PropTypes.func.isRequired,
+    activeBoundaries: PropTypes.object.isRequired,
+    getTooltipContentProps: PropTypes.func.isRequired
+  };
+
+  static getBoundariesItems(boundaries) {
+    return [
+      { value: null, label: 'No Boundaries' },
+      ...boundaries.map(boundary => ({
+        label: boundary.name,
+        value: { dataset: boundary.dataset, layers: [boundary.layer] }
+      }))
+    ];
+  }
+
+  state = {
```

Comment:
Do we need to be managing state and props if all is already in the props?

Reply:
yup you're right

Reply:
Do you still have time to address this?

Reply:
should be done, github is just being naughty

Reply:
😈